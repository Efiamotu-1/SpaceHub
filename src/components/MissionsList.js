import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Mission from './Mission';
import { asyncMissionsFromAPI } from '../redux/missions/missions';

const MissionsList = () => {
  const missionsArray = useSelector((state) => state.stMissions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(asyncMissionsFromAPI());
  }, [dispatch]);

  return (
    <ul className="ul-container">
      {missionsArray.map((mission) => (
        <Mission
          key={mission.id}
          mission={mission}
        />
      ))}
    </ul>
  );
};

export default MissionsList;
