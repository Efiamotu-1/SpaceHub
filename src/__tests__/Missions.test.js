import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configStore';
import MissionsList from '../components/MissionsList';

describe('Missions List Component', () => {
  test('Correct render of Missions List ', () => {
    const missions = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <MissionsList />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(missions).toMatchSnapshot();
  });
});
