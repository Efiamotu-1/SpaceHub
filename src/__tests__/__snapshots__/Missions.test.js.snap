// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions List Component Correct render of Missions List  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mission-container"
      >
        <div
          class="m_line"
        >
           
        </div>
        <div
          class="mu_header"
        >
          <div
            class="mu_mission"
          >
            <p>
              Mission
            </p>
          </div>
          <div
            class="mu_description"
          >
            <p>
              Description
            </p>
          </div>
          <div
            class="mu_statusAndButton"
          >
            <div
              class="mu_status"
            >
              <p>
                Status
              </p>
            </div>
            <div
              class="mu_button"
            >
               
            </div>
          </div>
        </div>
        <ul
          class="ul-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mission-container"
    >
      <div
        class="m_line"
      >
         
      </div>
      <div
        class="mu_header"
      >
        <div
          class="mu_mission"
        >
          <p>
            Mission
          </p>
        </div>
        <div
          class="mu_description"
        >
          <p>
            Description
          </p>
        </div>
        <div
          class="mu_statusAndButton"
        >
          <div
            class="mu_status"
          >
            <p>
              Status
            </p>
          </div>
          <div
            class="mu_button"
          >
             
          </div>
        </div>
      </div>
      <ul
        class="ul-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
